kube-controllers/pkg/controllers/node/ipam.go

```diff
@@ -568,16 +574,27 @@ func (c *ipamController) checkEmptyBlocks() error {
 			continue
 		}
 
+		// During a Flannel migration, we can only migrate blocks affined to nodes that have already undergone the migration
+		migrating, err := c.nodeIsBeingMigrated(node)
+		if err != nil {
+			logc.Warn("Couldn't find node affined to empty block, skipping affinity release")
```

Comment:
I don't think we can skip in this case - there are valid scenarios where blocks have been orphaned and the node no longer exists, and we need the garbage collector to properly release those blocks. We can only skip if the node exists and is actively undergoing a flannel migration.

Reply:
Then the helper should catch if the node doesn't exist and return that as well.
In case of a different error, we should assume that we just couldn't get the node from the API.?

Reply:
Yeah, I think it's OK to skip on other errors - those are true "error" scenarios. But NotFound shouldn't be treated as an error :+1:

Reply:
Sorry I missed this before - this log should include a `WithError` so we log out the original error.

e.g.,

```
logc.WithError(err).Warn("Failed to check if node is undergoing flannel migration")
```